force-app/main/default/classes/DMLOperations.cls

```diff
@@ -181,14 +212,16 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
+        for(Contact myContact : contactsToUpsert) {
+            // Extract the Account Name from Contact's LastName
+            String accountName = myContact.LastName;            
+             // Get the account based on the name or create one if it doesn't exist          
+           Account acc = upsertAccount(accountName);
```

Comment:
upsertAccount() does a DML upsert. this is inside of a for loop. What happens if this for loop has over 200 Contacts?

You have DML inside of a for loop.

Reply:
For this problem, you want to use maps to overcome the limitation here; I would say you need to refactor this problem so that you are getting the possible accounts, getting the possible contacts, and checking if they exist based on the input.